<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Gene expression in cells treated or not by a demethylating agent — DAC_induction • CTexploreR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gene expression in cells treated or not by a demethylating agent — DAC_induction"><meta name="description" content="Plots a heatmap of normalised gene counts
(log-transformed) in a selection of cells treated or not by
5-Aza-2'-Deoxycytidine (DAC), a demethylating agent."><meta property="og:description" content="Plots a heatmap of normalised gene counts
(log-transformed) in a selection of cells treated or not by
5-Aza-2'-Deoxycytidine (DAC), a demethylating agent."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CTexploreR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/CTexploreR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/UCLouvain-CBIO/CTexploreR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Gene expression in cells treated or not by a demethylating agent</h1>
      <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/CTexploreR/blob/main/R/DAC_induction.R" class="external-link"><code>R/DAC_induction.R</code></a></small>
      <div class="d-none name"><code>DAC_induction.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plots a heatmap of normalised gene counts
(log-transformed) in a selection of cells treated or not by
5-Aza-2'-Deoxycytidine (DAC), a demethylating agent.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">DAC_induction</span><span class="op">(</span></span>
<span>  genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  multimapping <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include_CTP <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  values_only <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p><code>character</code> naming the selected genes. The default
value, <code>NULL</code>, takes all CT specific genes.</p></dd>


<dt id="arg-multimapping">multimapping<a class="anchor" aria-label="anchor" href="#arg-multimapping"></a></dt>
<dd><p><code>logical(1)</code> defining whether to use
multi-mapped gene expression dataset
<code><a href="https://rdrr.io/pkg/CTdata/man/DAC_treated_cells_multimapping.html" class="external-link">CTdata::DAC_treated_cells_multimapping</a></code> or
<code>DAC_treated_cells</code>. Default is <code>TRUE</code>.</p></dd>


<dt id="arg-include-ctp">include_CTP<a class="anchor" aria-label="anchor" href="#arg-include-ctp"></a></dt>
<dd><p><code>logical(1)</code> If <code>TRUE</code>, CTP genes are included.
(<code>FALSE</code> by default).</p></dd>


<dt id="arg-values-only">values_only<a class="anchor" aria-label="anchor" href="#arg-values-only"></a></dt>
<dd><p><code>logical(1)</code>. If <code>TRUE</code>, the function will return the
gene normalised logcounts in all samples instead of the
heatmap. Default is <code>FALSE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A heatmap of selected genes in cells treated or not by a
demethylating agent. If <code>values_only</code> is <code>TRUE</code>, gene normalised
logcounts are returned instead.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>RNAseq data from cells treated or not with 5-aza downloaded from
SRA.  (SRA references and information about cell lines and DAC
treatment are stored the colData of <code>DAC_treated_cells</code>).  Data was
processed using a standard RNAseq pipeline.
<a href="https://ccb.jhu.edu/software/hisat2/index.shtml" class="external-link">hisat2</a> was used
to align reads to grch38 genome.
<a href="https://rdrr.io/bioc/Rsubread/man/featureCounts.html" class="external-link">featurecounts</a>
was used to assign reads to genes. Note that -M parameter was used
or not to allow or not counting multi-mapping reads.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">DAC_induction</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAGEA1"</span>, <span class="st">"MAGEA3"</span>, <span class="st">"MAGEA4"</span>, <span class="st">"MAGEA6"</span>, <span class="st">"CTAG1A"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?CTdata and browseVignettes('CTdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-plt img"><img src="DAC_induction-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">DAC_induction</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAGEA1"</span>, <span class="st">"MAGEA3"</span>, <span class="st">"MAGEA4"</span>, <span class="st">"MAGEA6"</span>, <span class="st">"CTAG1A"</span>,</span></span>
<span class="r-in"><span>    multimapping <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?CTdata and browseVignettes('CTdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>1 out of 6 names invalid: FALSE.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> See the manual page for valid types.</span>
<span class="r-plt img"><img src="DAC_induction-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Axelle Loriot, Julie Devis, Laurent Gatto.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer></div>





  </body></html>

