<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Gene expression in cells treated or not by a demethylating agent — DAC_induction • CTexploreR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Gene expression in cells treated or not by a demethylating agent — DAC_induction"><meta property="og:description" content="Plots a heatmap of normalised gene counts
(log-transformed) in a selection of cells treated or not by
5-Aza-2'-Deoxycytidine (DAC), a demethylating agent."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CTexploreR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CTexploreR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/UCLouvain-CBIO/CTexploreR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene expression in cells treated or not by a demethylating agent</h1>
    <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/CTexploreR/blob/HEAD/R/DAC_induction.R" class="external-link"><code>R/DAC_induction.R</code></a></small>
    <div class="hidden name"><code>DAC_induction.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plots a heatmap of normalised gene counts
(log-transformed) in a selection of cells treated or not by
5-Aza-2'-Deoxycytidine (DAC), a demethylating agent.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">DAC_induction</span><span class="op">(</span>genes <span class="op">=</span> <span class="cn">NULL</span>, multimapping <span class="op">=</span> <span class="cn">TRUE</span>, return <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genes</dt>
<dd><p><code>character</code> nameing the selected genes. The default
value, <code>NULL</code>, takes all CT genes.</p></dd>


<dt>multimapping</dt>
<dd><p><code>logical(1)</code> defining whether to use
multi-mapped gene expression dataset
<code><a href="https://rdrr.io/pkg/CTdata/man/DAC_treated_cells_multimapping.html" class="external-link">CTdata::DAC_treated_cells_multimapping</a></code> or
<code>DAC_treated_cells</code>. Default is <code>TRUE</code>.</p></dd>


<dt>return</dt>
<dd><p><code>logical(1)</code>. If <code>TRUE</code>, the function will return the
gene normalised logcounts in all samples instead of the
heatmap. Default is <code>FALSE</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A heatmap of selected genes in cells treated or not by a
demethylating agent. If return is <code>TRUE</code>, gene normalised
logcounts are returned instead.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>RNAseq data from cells treated or not with 5-aza downloaded from
SRA.  (SRA references and information about cell lines and DAC
treatment are stored the colData of <code>DAC_treated_cells</code>).  Data was
processed using a standard RNAseq pipeline.
<a href="https://ccb.jhu.edu/software/hisat2/index.shtml" class="external-link">hisat2</a> was used
to align reads to grch38 genome.
<a href="https://rdrr.io/bioc/Rsubread/man/featureCounts.html" class="external-link">featurecounts</a>
was used to assign reads to genes. Note that -M parameter was used
or not to allow or not counting multi-mapping reads.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">DAC_induction</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAGEA1"</span>, <span class="st">"MAGEA3"</span>, <span class="st">"MAGEA4"</span>, <span class="st">"MAGEA6"</span>, <span class="st">"CTAG1A"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="DAC_induction-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">DAC_induction</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAGEA1"</span>, <span class="st">"MAGEA3"</span>, <span class="st">"MAGEA4"</span>, <span class="st">"MAGEA6"</span>, <span class="st">"CTAG1A"</span>,</span></span>
<span class="r-in"><span>    multimapping <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>1 out of 6 names invalid: FALSE.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> See the manual page for valid types.</span>
<span class="r-plt img"><img src="DAC_induction-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Axelle Loriot, Julie Devis, Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

