<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Expression values (TPM) of genes in normal tissues with or without multimapping — normal_tissue_expression_multimapping • CTexploreR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Expression values (TPM) of genes in normal tissues with or without multimapping — normal_tissue_expression_multimapping"><meta name="description" content="Plots a heatmap of gene expression values in a set of normal
tissues. Expression values (in TPM) have been evaluated by either
counting or discarding multi-mapped reads. Indeed, many CT genes
belong to gene families from which members have identical or nearly
identical sequences. Some CT can only be detected in RNAseq data in
which multimapping reads are not discarded."><meta property="og:description" content="Plots a heatmap of gene expression values in a set of normal
tissues. Expression values (in TPM) have been evaluated by either
counting or discarding multi-mapped reads. Indeed, many CT genes
belong to gene families from which members have identical or nearly
identical sequences. Some CT can only be detected in RNAseq data in
which multimapping reads are not discarded."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CTexploreR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/CTexploreR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/UCLouvain-CBIO/CTexploreR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Expression values (TPM) of genes in normal tissues with or without multimapping</h1>
      <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/CTexploreR/blob/main/R/normal_tissue_expression_multimapping.R" class="external-link"><code>R/normal_tissue_expression_multimapping.R</code></a></small>
      <div class="d-none name"><code>normal_tissue_expression_multimapping.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plots a heatmap of gene expression values in a set of normal
tissues. Expression values (in TPM) have been evaluated by either
counting or discarding multi-mapped reads. Indeed, many CT genes
belong to gene families from which members have identical or nearly
identical sequences. Some CT can only be detected in RNAseq data in
which multimapping reads are not discarded.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">normal_tissue_expression_multimapping</span><span class="op">(</span></span>
<span>  genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  include_CTP <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  multimapping <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  units <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TPM"</span>, <span class="st">"log_TPM"</span><span class="op">)</span>,</span>
<span>  values_only <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p><code>character</code> naming the selected genes. The default
value, <code>NULL</code>, takes all CT (specific) genes.</p></dd>


<dt id="arg-include-ctp">include_CTP<a class="anchor" aria-label="anchor" href="#arg-include-ctp"></a></dt>
<dd><p><code>logical(1)</code> If <code>TRUE</code>, CTP genes are included.
(<code>FALSE</code> by default).</p></dd>


<dt id="arg-multimapping">multimapping<a class="anchor" aria-label="anchor" href="#arg-multimapping"></a></dt>
<dd><p><code>logical(1)</code> that specifies if returned
expression values must take into account or not multi-mapped
reads. TRUE by default.</p></dd>


<dt id="arg-units">units<a class="anchor" aria-label="anchor" href="#arg-units"></a></dt>
<dd><p><code>character(1)</code> with expression values unit.  Can be
<code>"TPM"</code> (default) or <code>"log_TPM"</code> (log(TPM + 1)).</p></dd>


<dt id="arg-values-only">values_only<a class="anchor" aria-label="anchor" href="#arg-values-only"></a></dt>
<dd><p><code>logical(1)</code>. If <code>TRUE</code>, the function will return the
expression values in all samples instead of the
heatmap. Default is <code>FALSE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A heatmap of selected gene expression values in a set of
normal tissues calculated by counting or discarding
multi-mapped reads.  If <code>values_only = TRUE</code>, gene expression values
are returned instead.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>RNAseq data from a set of normal tissues were downloaded from
Encode.  (see inst/scripts/make_CT_normal_tissues_multimapping.R
for fastq references) Fastq files were processed using a standard
RNAseq pipeline including
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="external-link">FastQC</a>
for the quality control of the raw data, and
<a href="http://www.usadellab.org/cms/?page=trimmomatic" class="external-link">trimmomatic</a> to
remove low quality reads and trim the adapter from the sequences.
<a href="https://ccb.jhu.edu/software/hisat2/index.shtml" class="external-link">hisat2</a> was used
to align reads to grch38 genome.
<a href="https://rdrr.io/bioc/Rsubread/man/featureCounts.html" class="external-link">featurecounts</a>
was used to assign reads to genes using
Homo_sapiens.GRCh38.105.gtf.</p>
<p>Two different pipelines were run in order to remove or not
multi-mapping reads.  When multimapping was allowed, hisat2 was run
with -k 20 parameter (reports up to 20 alignments per read), and
featurecounts was run with -M parameter (multi-mapping reads are
counted).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">normal_tissue_expression_multimapping</span><span class="op">(</span></span></span>
<span class="r-in"><span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GAGE13"</span>, <span class="st">"CT45A6"</span>, <span class="st">"NXF2"</span>, <span class="st">"SSX2"</span>, <span class="st">"CTAG1A"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"MAGEA3"</span>, <span class="st">"MAGEA6"</span><span class="op">)</span>, multimapping <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?CTdata and browseVignettes('CTdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-plt img"><img src="normal_tissue_expression_multimapping-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">normal_tissue_expression_multimapping</span><span class="op">(</span></span></span>
<span class="r-in"><span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GAGE13"</span>, <span class="st">"CT45A6"</span>, <span class="st">"NXF2"</span>, <span class="st">"SSX2"</span>, <span class="st">"CTAG1A"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"MAGEA3"</span>, <span class="st">"MAGEA6"</span><span class="op">)</span>, multimapping <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?CTdata and browseVignettes('CTdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-plt img"><img src="normal_tissue_expression_multimapping-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Axelle Loriot, Julie Devis, Laurent Gatto.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer></div>





  </body></html>

