<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Cancer Testis explorer • CTexploreR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Cancer Testis explorer">
<meta property="og:description" content="CTexploreR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CTexploreR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CTexploreR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/UCLouvain-CBIO/CTexploreR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Cancer Testis explorer</h1>
                        <h4 data-toc-skip class="author">Julie Devis,
Laurent Gatto, Axelle Loriot</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/CTexploreR/blob/HEAD/vignettes/CTexploreR.Rmd" class="external-link"><code>vignettes/CTexploreR.Rmd</code></a></small>
      <div class="hidden name"><code>CTexploreR.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Cancer-Testis (CT) genes, also called Cancer-Germline (CG), are a
group of genes whose expression is normally restricted to the germline
but that are found aberrantly activated in many types of cancers. These
genes produce cancer specific antigens represent ideal targets for
anti-cancer vaccines. Besides their interest in immunotherapy, they can
also be used as cancer biomarkers and as target of anti-tumor therapies
with limited side effect.</p>
<p>Many CT genes use DNA methylation as a primary mechanism of
transcriptionnal regulation. This is another interesting point about CT
genes as they represent suitable models to study DNA demethylation in
cancer.</p>
<p>Currently the reference database gathering CT genes is the <a href="http://www.cta.lncc.br/" class="external-link">CTdatabase</a> that was published in
2009, based on a literature screening(Almeida et al., 2009). This
database is however not up to date. Recently identified CT genes are not
referenced (in particular CT genes identified by omics methods that
didn’t exist at the time) while some genes referred as CT genes appeared
to be in reality expressed in many somatic tissues. Furthermore, the
database is not in an easily importable format, some genes are not
encoded properly (by synonyms names rather than by their official HGNC
symbol names, or by a concatenation of both) resulting in poor
interoperability for downstream analyses. More recent studies proposed
other lists of CT genes like Wang’s CTatlas (Wang et al., 2016, Jamin et
al., 2021, Carter et al., 2023). These lists were established using
different criteria to define CT genes and hence differ substantially
from each other. Moreover, these lists are usually provided as
supplemental data files and are not strictly speaking databases.
Finally, none of these studies describe the involvement of DNA
methylation in the regulation of individual CT genes.</p>
<p>We therefore created <code>CTexploreR</code>, a Bioconductor R
package, aimed to redefine rigorously the list of CT genes based on
publicly available RNAseq databases and to summarize their main
characteristics. We included methylation analyses to classify these
genes according to whether they are regulated or not by DNA methylation.
The package also offers tools to visualize CT genes expression and
promoter DNA methylation in normal and tumoral tissues. CTexploreR hence
represents an up-to-date reference database for CT genes and can be used
as a starting point for further investigations related to these
genes.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"CTexploreR"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"CTexploreR"</span><span class="op">)</span></span></code></pre></div>
<p>To install the package from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"UCLouvain-CBIO/CTexploreR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ct-genes">CT genes<a class="anchor" aria-label="anchor" href="#ct-genes"></a>
</h2>
<p>The central element of <code>CTexploreR</code> is the list of 298 CT
genes (see table below) selected based on their expression in normal and
tumoral tissues (selection details in the next section). The table also
summarises their main characteristics.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/UCLouvain-CBIO/CTexploreR" class="external-link">CTexploreR</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: CTdata</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CT_genes</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 298 × 36</span></span></span>
<span><span class="co">##    ensembl_gene_id external_gene_name family chr   strand transcription_start_…¹</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ENSG00000162571 TTLL10             <span style="color: #BB0000;">NA</span>     1          1                1<span style="text-decoration: underline;">173</span>880</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ENSG00000173662 TAS1R1             <span style="color: #BB0000;">NA</span>     1          1                6<span style="text-decoration: underline;">555</span>307</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ENSG00000157330 CFAP107            <span style="color: #BB0000;">NA</span>     1          1               12<span style="text-decoration: underline;">746</span>200</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ENSG00000234593 KAZN-AS1           <span style="color: #BB0000;">NA</span>     1         -<span style="color: #BB0000;">1</span>               14<span style="text-decoration: underline;">419</span>973</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ENSG00000117148 ACTL8              <span style="color: #BB0000;">NA</span>     1          1               17<span style="text-decoration: underline;">755</span>333</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ENSG00000131914 LIN28A             <span style="color: #BB0000;">NA</span>     1          1               26<span style="text-decoration: underline;">410</span>817</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ENSG00000142698 C1orf94            <span style="color: #BB0000;">NA</span>     1          1               34<span style="text-decoration: underline;">176</span>907</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ENSG00000143006 DMRTB1             <span style="color: #BB0000;">NA</span>     1          1               53<span style="text-decoration: underline;">459</span>399</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ENSG00000237853 NFIA-AS1           <span style="color: #BB0000;">NA</span>     1         -<span style="color: #BB0000;">1</span>               61<span style="text-decoration: underline;">253</span>510</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ENSG00000226088 HHLA3-AS1          <span style="color: #BB0000;">NA</span>     1         -<span style="color: #BB0000;">1</span>               70<span style="text-decoration: underline;">360</span>437</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 288 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ abbreviated name: ¹​transcription_start_site</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 30 more variables: X_linked &lt;lgl&gt;, TPM_testis &lt;dbl&gt;, max_TPM_somatic &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   GTEX_category &lt;chr&gt;, lowly_expressed_in_GTEX &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   multimapping_analysis &lt;chr&gt;, testis_specificity &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   testis_cell_type &lt;chr&gt;, Higher_in_somatic_cell_type &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   percent_of_positive_CCLE_cell_lines &lt;dbl&gt;, …</span></span></span></code></pre>
<p><code>CTdata</code> is the companion Package for
<code>CTexploreR</code> and provides the omics data that was necessary
to select and characterize cancer testis genes as well as exploring
them. The data are served through the <code>ExperimentHub</code>
infrastructure. Currently available data are summarised in the table
below and details can be found in <code>CTdata</code> vignette or
manuals.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/CTdata/man/CTdata.html" class="external-link">CTdata</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                               Title</span></span>
<span><span class="co">## 1                         GTEX_data</span></span>
<span><span class="co">## 2                         CCLE_data</span></span>
<span><span class="co">## 3  normal_tissues_multimapping_data</span></span>
<span><span class="co">## 4                 DAC_treated_cells</span></span>
<span><span class="co">## 5    DAC_treated_cells_multimapping</span></span>
<span><span class="co">## 6                          TCGA_TPM</span></span>
<span><span class="co">## 7         CT_methylation_in_tissues</span></span>
<span><span class="co">## 8    CT_mean_methylation_in_tissues</span></span>
<span><span class="co">## 9               TCGA_CT_methylation</span></span>
<span><span class="co">## 10                         CT_genes</span></span>
<span><span class="co">## 11          CCLE_correlation_matrix</span></span>
<span><span class="co">## 12                       testis_sce</span></span>
<span><span class="co">## 13                     scRNAseq_HPA</span></span>
<span><span class="co">##                                                                                                              Description</span></span>
<span><span class="co">## 1                                                              Gene expression data in normal tissues from GTEx database</span></span>
<span><span class="co">## 2                                                                    Gene expression data in cancer cell lines from CCLE</span></span>
<span><span class="co">## 3                                         Gene expression values in normal tissues with or without allowing multimapping</span></span>
<span><span class="co">## 4  Gene expression values in a set of cell lines treated or not with 5-Aza-2'-Deoxycytidine (DAC), a demethylating agent</span></span>
<span><span class="co">## 5  Gene expression values (multimapping allowed) in a set of cell lines treated or not with 5-Aza-2'-Deoxycytidine (DAC)</span></span>
<span><span class="co">## 6                                                                                   Gene expression data in TCGA samples</span></span>
<span><span class="co">## 7                           Methylation values of CpGs located within Cancer-Testis promoters in a set of normal tissues</span></span>
<span><span class="co">## 8             Mean methylation values of all CpGs located within Cancer-Testis (CT) promoters in a set of normal tissues</span></span>
<span><span class="co">## 9                                                                            Methylation of CT promoters in TCGA samples</span></span>
<span><span class="co">## 10                                                                                  Cancer-Testis (CT) genes description</span></span>
<span><span class="co">## 11                                                                           Gene correlations in CCLE cancer cell lines</span></span>
<span><span class="co">## 12                                                                                                  Testis scRNAseq data</span></span>
<span><span class="co">## 13            Gene expression profiles in different human cell types based on scRNAseq data from The Human Protein Atlas</span></span>
<span><span class="co">##    BiocVersion Genome SourceType</span></span>
<span><span class="co">## 1         3.17   hg38        tab</span></span>
<span><span class="co">## 2         3.17   hg38        CSV</span></span>
<span><span class="co">## 3         3.17   hg38      FASTQ</span></span>
<span><span class="co">## 4         3.17   hg38      FASTQ</span></span>
<span><span class="co">## 5         3.17   hg38      FASTQ</span></span>
<span><span class="co">## 6         3.17   hg38        TSV</span></span>
<span><span class="co">## 7         3.17   hg38        BED</span></span>
<span><span class="co">## 8         3.17   hg38        BED</span></span>
<span><span class="co">## 9         3.17   hg38        TXT</span></span>
<span><span class="co">## 10        3.17   hg38        TSV</span></span>
<span><span class="co">## 11        3.17   hg38        CSV</span></span>
<span><span class="co">## 12        3.18   hg38      FASTQ</span></span>
<span><span class="co">## 13        3.18   hg38        TSV</span></span>
<span><span class="co">##                                                                                                                        SourceUrl</span></span>
<span><span class="co">## 1  https://storage.googleapis.com/gtex_analysis_v8/rna_seq_data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz</span></span>
<span><span class="co">## 2                               https://ndownloader.figshare.com/files/34989922, https://ndownloader.figshare.com/files/35020903</span></span>
<span><span class="co">## 3                                                                                                 https://www.encodeproject.org/</span></span>
<span><span class="co">## 4                                                                                                 https://www.encodeproject.org/</span></span>
<span><span class="co">## 5                                                                                                 https://www.encodeproject.org/</span></span>
<span><span class="co">## 6                                                                                                 https://portal.gdc.cancer.gov/</span></span>
<span><span class="co">## 7                                                                                                 https://www.encodeproject.org/</span></span>
<span><span class="co">## 8                                                                                                 https://www.encodeproject.org/</span></span>
<span><span class="co">## 9                                                                                                 https://portal.gdc.cancer.gov/</span></span>
<span><span class="co">## 10                                                    https://zenodo.org/record/7537824/files/cancermine_collated.tsv?download=1</span></span>
<span><span class="co">## 11                              https://ndownloader.figshare.com/files/34989922, https://ndownloader.figshare.com/files/35020903</span></span>
<span><span class="co">## 12                                                                  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120508</span></span>
<span><span class="co">## 13                                                            https://www.proteinatlas.org/download/rna_single_cell_type.tsv.zip</span></span>
<span><span class="co">##    SourceVersion      Species TaxonomyId Coordinate_1_based</span></span>
<span><span class="co">## 1              8 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 2           22Q2 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 3              5 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 4              5 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 5              5 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 6           36.0 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 7              5 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 8              5 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 9           36.0 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 10            49 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 11          22Q2 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 12             1 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">## 13             1 Homo sapiens       9606               TRUE</span></span>
<span><span class="co">##                                                          DataProvider</span></span>
<span><span class="co">## 1                                      The Genotype-Tissue Expression</span></span>
<span><span class="co">## 2                                       Cancer Cell Line Encyclopedia</span></span>
<span><span class="co">## 3                                                      ENCODE Project</span></span>
<span><span class="co">## 4                                                      ENCODE Project</span></span>
<span><span class="co">## 5                                                      ENCODE Project</span></span>
<span><span class="co">## 6                                             The Cancer Genome Atlas</span></span>
<span><span class="co">## 7                                                      ENCODE Project</span></span>
<span><span class="co">## 8                                                      ENCODE Project</span></span>
<span><span class="co">## 9                                             The Cancer Genome Atlas</span></span>
<span><span class="co">## 10                                                         CancerMine</span></span>
<span><span class="co">## 11                                      Cancer Cell Line Encyclopedia</span></span>
<span><span class="co">## 12 The adult human testis transcriptional cell atlas, Guo et al. 2018</span></span>
<span><span class="co">## 13                                            The Human Protein Atlas</span></span>
<span><span class="co">##                                    Maintainer                 RDataClass</span></span>
<span><span class="co">## 1  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 2  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 3  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 4  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 5  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 6  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 7  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt; RangedSummarizedExperiment</span></span>
<span><span class="co">## 8  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SummarizedExperiment</span></span>
<span><span class="co">## 9  Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt; RangedSummarizedExperiment</span></span>
<span><span class="co">## 10 Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;                     tibble</span></span>
<span><span class="co">## 11 Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;                     matrix</span></span>
<span><span class="co">## 12 Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SingleCellExperiment</span></span>
<span><span class="co">## 13 Axelle Loriot &lt;axelle.loriot@uclouvain.be&gt;       SingleCellExperiment</span></span>
<span><span class="co">##    DispatchClass                                            RDataPath</span></span>
<span><span class="co">## 1            Rda                         CTdata/eh_data/GTEX_data.rda</span></span>
<span><span class="co">## 2            Rda                         CTdata/eh_data/CCLE_data.rda</span></span>
<span><span class="co">## 3            Rda  CTdata/eh_data/normal_tissues_multimapping_data.rda</span></span>
<span><span class="co">## 4            Rda                 CTdata/eh_data/DAC_treated_cells.rda</span></span>
<span><span class="co">## 5            Rda    CTdata/eh_data/DAC_treated_cells_multimapping.rda</span></span>
<span><span class="co">## 6            Rda                          CTdata/eh_data/TCGA_TPM.rda</span></span>
<span><span class="co">## 7            Rda      CTdata/eh_data/v2/CT_methylation_in_tissues.rda</span></span>
<span><span class="co">## 8            Rda CTdata/eh_data/v2/CT_mean_methylation_in_tissues.rda</span></span>
<span><span class="co">## 9            Rda            CTdata/eh_data/v2/TCGA_CT_methylation.rda</span></span>
<span><span class="co">## 10           Rda                       CTdata/eh_data/v2/CT_genes.rda</span></span>
<span><span class="co">## 11           Rda        CTdata/eh_data/v2/CCLE_correlation_matrix.rda</span></span>
<span><span class="co">## 12           Rda                        CTdata/eh_data/testis_sce.rda</span></span>
<span><span class="co">## 13           Rda                      CTdata/eh_data/scRNAseq_HPA.rda</span></span>
<span><span class="co">##                                                                                                                                                                    Tags</span></span>
<span><span class="co">## 1                                                       ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:NormalTissues</span></span>
<span><span class="co">## 2                                                ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:CancerData:CellLines</span></span>
<span><span class="co">## 3                                          ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:NormalTissues:Multimapping</span></span>
<span><span class="co">## 4                                        ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:CellLines:DemethylatingAgent</span></span>
<span><span class="co">## 5                           ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:CellLines:DemethylatingAgent:Multimapping</span></span>
<span><span class="co">## 6                                             ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:CancerData:TumorSamples</span></span>
<span><span class="co">## 7                                         ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:MethylationData:CancerTestis:NormalTissues</span></span>
<span><span class="co">## 8                                         ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:MethylationData:CancerTestis:NormalTissues</span></span>
<span><span class="co">## 9                               ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:MethylationData:CancerTestis:CancerData:TumorSamples</span></span>
<span><span class="co">## 10 ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:CancerTestis:MethylationData:GeneExpression:CancerData:NormalTissues:TumorSamples</span></span>
<span><span class="co">## 11                                  ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:CancerTestis:GeneExpression:CancerData:CellLines</span></span>
<span><span class="co">## 12                                                    ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:scRNAseq:testis</span></span>
<span><span class="co">## 13                                           ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:Homo_sapiens_Data:GeneExpression:scRNAseq:normalCellTypes</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="ct-gene-selection">CT gene selection<a class="anchor" aria-label="anchor" href="#ct-gene-selection"></a>
</h2>
<p>In order to generate the list of CT genes, we followed a specific
selection procedure (see figure below).</p>
<p><img src="CT_selection.svg" width="100%" style="display: block; margin: auto;"></p>
<div class="section level3">
<h3 id="testis-specific-expression">Testis-specific expression<a class="anchor" aria-label="anchor" href="#testis-specific-expression"></a>
</h3>
<p>Testis-specific genes (expressed exclusively in testis) and
testis-preferential genes (expressed in a few somatic tissues at a level
lower than 10x testis expression) were first selected using the GTEx
database (Aguet et al., 2020).</p>
<p>Note that some genes were undetectable in the GTEx database due to
multimapping issues (these were flagged as “lowly expressed” in
<code>GTEX_category</code> column). A careful inspection of these genes
showed that many of them are well-known Cancer-Testis genes, belonging
to gene families (MAGEA, SSX, CT45A, GAGE, …) from which members have
identical or nearly identical sequences. This is likely the reason why
these genes are not detected in the GTEx database, as GTEx processing
pipeline specifies that overlapping intervals between genes are excluded
from all genes for counting.For these genes, as testis-specificity could
not be assessed using GTEx database, RNAseq data from a set of normal
tissues were reprocessed in order to allow multimapping. Expression of
several genes became detectable in some tissues. Genes showing a
testis-specific expression (expression at least 10x higher in testis
than in any somatic tissues when multimapping was allowed) were
selected, and flagged as testis-specific in
<code>multimapping_analysis</code> column.</p>
<p>The expression of the selected testis-specific or testis-preferential
genes was further analysed in somatic cell types using scRNAseq data of
normal tissues from the Human Protein Atlas (Uhlén et al., 2015). The
aim was to ensure that the selected genes are not expressed in any
specific somatic cell type, as the GTEx selection was based on bulk
RNAseq data.</p>
</div>
<div class="section level3">
<h3 id="germline-specific-expression">Germline-specific expression<a class="anchor" aria-label="anchor" href="#germline-specific-expression"></a>
</h3>
<p>We also used the single cell RNA-Seq data from the adult human testis
transcriptional cell atlas (Guo et al., 2018) to ensure to select
germline-specific genes but not genes that would be specific of any
somatic cell type of the testis.</p>
<p>For each gene, the testis cell type has been determined as the cell
type showing the highest mean expression for that gene. This allowed us
to remove the genes for which this cell type corresponds to a testis
somatic cell type (macrophage, endothelial, myoid, Sertoli or Leydig
cells).</p>
</div>
<div class="section level3">
<h3 id="activation-in-cancer-cell-lines-and-tcga-tumors">Activation in cancer cell lines and TCGA tumors<a class="anchor" aria-label="anchor" href="#activation-in-cancer-cell-lines-and-tcga-tumors"></a>
</h3>
<p>To assess activation in cancers, RNAseq data from cancer cell lines
from CCLE (Barretina et al., 2012) and from TCGA cancer samples (Cancer
Genome Atlas Research Network et al., 2013) were used. This allowed to
select among testis-specific and testis-preferential genes those that
are activated in cancers.</p>
<p>In the <code>CCLE_category</code> and <code>TCGA_category</code>
columns, genes are tagged as “activated” when they are highly expressed
in at least one cancer cell line/ sample (TPM &gt;= 10). However genes
that were found to be expressed in all -or almost all-cancer cell
lines/samples were removed, as this probably reflects a constitutive
expression rather than a true activation. We filtered out genes that
were not completely repressed in at least 20 % of cancer cell
lines/samples (TPM &lt;= 0.1).</p>
</div>
<div class="section level3">
<h3 id="igv-visualisation">IGV visualisation<a class="anchor" aria-label="anchor" href="#igv-visualisation"></a>
</h3>
<p>All selected CT genes were visualised on IGV (Thorvaldsdóttir et al.,
2013) using a RNA-seq alignment from testis, to ensure that expression
in testis really corresponded to the canonical transcript. For some
genes for which the canonical transcript did not correspond to the
transcript that we could see in the testis sample, we manually modified
the <code>external_transcript_name</code> accordingly, to ensure that
the TSS and the promoter region are correctly defined. This is
particularly important for methylation analysis that must be focused on
true promoter regions.</p>
</div>
<div class="section level3">
<h3 id="regulation-by-methylation">Regulation by methylation<a class="anchor" aria-label="anchor" href="#regulation-by-methylation"></a>
</h3>
<p>Genes flagged as <code>TRUE</code> in
<code>regulated_by_methylation</code> column correspond to</p>
<ul>
<li><p>Genes that are significantly induced by a demethylating agent
(RNAseq analysis of cell lines reated with DAC
(5-Aza-2′-Deoxycytidine)).</p></li>
<li><p>Genes that have a highly methylated promoter in normal somatic
tissues but less methylated in germ cells (WGBS analysis of a set of
normal tissues).</p></li>
</ul>
<p>For some genes showing a strong activation in cells treated with
5-Aza-2′-Deoxycytidine, methylation analysis was not possible due to
multimapping issues. In this case, genes were still considered as
regulated by methylation unless their promoter appeared unmethylated in
somatic tissues or methylated in germ cells.</p>
</div>
</div>
<div class="section level2">
<h2 id="available-functions">Available functions<a class="anchor" aria-label="anchor" href="#available-functions"></a>
</h2>
<p>For details about functions, see their respective manual pages. For
all functions, an option <code>return</code> can be set to
<code>TRUE</code> in order to get the values instead of the
visualisation.</p>
<p>All expression visualisation functions can be used on all GTEx genes,
not only on Cancer-Testis genes, as the data they refer to contains all
genes.</p>
<div class="section level3">
<h3 id="expression-in-normal-healthy-tissues">Expression in normal healthy tissues<a class="anchor" aria-label="anchor" href="#expression-in-normal-healthy-tissues"></a>
</h3>
<div class="section level4">
<h4 id="gtex_expression">
<code>GTEX_expression()</code><a class="anchor" aria-label="anchor" href="#gtex_expression"></a>
</h4>
<p>Allows to visualise gene expression in GTEx tissues. We can for
example see the difference of expression between testis-specific and
testis-preferential genes. Testis-specific genes have been determined
with a stricter specificity to the testis : they are lowly expressed in
all somatic tissues and at least 10 times more in the testis. Whereas
testis-preferential accepts a little expression outside the testis :
they are lowly expressed in at least 75% of somatic tissues, but still
10 times more in the testis.</p>
<ul>
<li>Applied to testis-specific genes : we can clearly see the expression
strictly limited to the testis. We can also see genes that are lowly
expressed in GTEx, and have thus been characterized using multimapping
(see below).</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testis_specific</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">CT_genes</span>,</span>
<span>    <span class="va">testis_specificity</span> <span class="op">==</span> <span class="st">"testis_specific"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/GTEX_expression.html">GTEX_expression</a></span><span class="op">(</span><span class="va">testis_specific</span><span class="op">$</span><span class="va">external_gene_name</span>, units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<ul>
<li>Applied to testis-preferential genes : we can see that the
expression is less stringent to testis, as expected, with low expression
in some other tissues, always with a strong testis signal.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testis_preferential</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">CT_genes</span>, <span class="va">testis_specificity</span> <span class="op">==</span> <span class="st">"testis_preferential"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/GTEX_expression.html">GTEX_expression</a></span><span class="op">(</span><span class="va">testis_preferential</span><span class="op">$</span><span class="va">external_gene_name</span>, units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="normal_tissue_expression_multimapping">
<code>normal_tissue_expression_multimapping()</code><a class="anchor" aria-label="anchor" href="#normal_tissue_expression_multimapping"></a>
</h4>
<p>Allows to visualise the expression values obtained by counting or not
multi-mapped reads in normal tissues. We can apply that to genes that
are lowly expressed in GTEx (as seen above) in order to visualise their
expression. First heatmap shows expression without multimapping while
multimapping was allowed in the second one, where expression can be
observed. This allowed their determination as testis-specific genes.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testis_specific_in_multimapping_analysis</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CT_genes</span>, <span class="va">lowly_expressed_in_GTEX</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/normal_tissue_expression_multimapping.html">normal_tissue_expression_multimapping</a></span><span class="op">(</span></span>
<span>    <span class="va">testis_specific_in_multimapping_analysis</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    multimapping <span class="op">=</span> <span class="cn">FALSE</span>, units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/normal_tissue_expression_multimapping.html">normal_tissue_expression_multimapping</a></span><span class="op">(</span></span>
<span>    <span class="va">testis_specific_in_multimapping_analysis</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    multimapping <span class="op">=</span> <span class="cn">TRUE</span>, units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-4-2.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="testis_expression">
<code>testis_expression()</code><a class="anchor" aria-label="anchor" href="#testis_expression"></a>
</h4>
<p>Allows to visualise gene expression in all different testis cell
type, somatic or germ, using data from the adult human testis
transcriptional cell atlas.</p>
<p>Using the testis cell type defined for each CT genes, we visualise
clearly that genes mainly expressed in an early stage of spermatogenesis
aren’t expressed later and vice-versa.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">early_CT</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CT_genes</span>, <span class="va">testis_cell_type</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span></span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SSC"</span>, <span class="st">"Spermatogonia"</span>, <span class="st">"Early_spermatocyte"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/testis_expression.html">testis_expression</a></span><span class="op">(</span><span class="va">early_CT</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>                  cells <span class="op">=</span> <span class="st">"germ_cells"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">## 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">## setting TRUE/FALSE to it.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Set `ht_opt$message = FALSE` to turn off this message.</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">late_CT</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CT_genes</span>, <span class="va">testis_cell_type</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span></span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Late_spermatocyte"</span>, <span class="st">"Round_spermatocyte"</span>,</span>
<span>                      <span class="st">"Sperm1"</span>, <span class="st">"Sperm2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/testis_expression.html">testis_expression</a></span><span class="op">(</span><span class="va">late_CT</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>                  cells <span class="op">=</span> <span class="st">"germ_cells"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">## 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">## setting TRUE/FALSE to it.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Set `ht_opt$message = FALSE` to turn off this message.</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="hpa_cell_type_expression">
<code>HPA_cell_type_expression()</code><a class="anchor" aria-label="anchor" href="#hpa_cell_type_expression"></a>
</h4>
<p>Allows to visualise gene expression in all different healthy cell
type, comparing different somatic cells to germ cells, using data from
the human proteome atlas.</p>
<p>Visualising all CT_genes, the specificity to germ cells only is quite
clear.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/HPA_cell_type_expression.html">HPA_cell_type_expression</a></span><span class="op">(</span>units <span class="op">=</span> <span class="st">"scaled"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?CTdata and browseVignettes('CTdata') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Warning: 48 out of 298 names invalid: KAZN-AS1, NFIA-AS1, HHLA3-AS1, SPATA42,</span></span>
<span><span class="co">## LINC01356, LINC01940, NECTIN3-AS1, AADACL2-AS1, LINC01206, LINC01096,</span></span>
<span><span class="co">## LINC02475, LINC02377, LINC01095, LINC01511, LINC02241, LINC02228,</span></span>
<span><span class="co">## LINC02105, LINC01947, SLIT3-AS1, MIR548A1HG, LINC01016, TDRG1,</span></span>
<span><span class="co">## LINC03004, SEC61G-DT, GTF2I-AS1, NRG1-IT1, IKBKB-DT, LINC01030,</span></span>
<span><span class="co">## LINC00200, LINC01518, MRGPRG-AS1, TMEM132D-AS1, FAM230C, LINC00408,</span></span>
<span><span class="co">## LINC01467, LINC01193, LINC01413, CT62, LINC02169, FLJ36000, LINC02074,</span></span>
<span><span class="co">## LINC01970, LINC00470, LINC00668, LINC02864, LINC01548, DSCR8, FAM230I.</span></span>
<span><span class="co">## See the manual page for valid types.</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="expression-in-cancer-cells-and-samples">Expression in cancer cells and samples<a class="anchor" aria-label="anchor" href="#expression-in-cancer-cells-and-samples"></a>
</h3>
<div class="section level4">
<h4 id="ccle_expression">
<code>CCLE_expression()</code><a class="anchor" aria-label="anchor" href="#ccle_expression"></a>
</h4>
<p>Allows to visualise gene expression in different histological types
of CCLE cancer cell lines. We can thus compare genes that are or not
activated in tumoral cell lines.</p>
<ul>
<li>Applied to CT genes frequently activated in CCLE cell lines, more
than 5% of CCLE cell_lines are expressing each gene. A string signal is
visible.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">frequently_activated</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">CT_genes</span>,</span>
<span>    <span class="va">percent_of_positive_CCLE_cell_lines</span> <span class="op">&gt;=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/CCLE_expression.html">CCLE_expression</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">frequently_activated</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"lung"</span>, <span class="st">"skin"</span>, <span class="st">"bile_duct"</span>, <span class="st">"bladder"</span>, <span class="st">"colorectal"</span>,</span>
<span>        <span class="st">"lymphoma"</span>, <span class="st">"uterine"</span>, <span class="st">"myeloma"</span>, <span class="st">"kidney"</span>,</span>
<span>        <span class="st">"pancreatic"</span>, <span class="st">"brain"</span>, <span class="st">"gastric"</span>, <span class="st">"breast"</span>, <span class="st">"bone"</span>,</span>
<span>        <span class="st">"head_and_neck"</span>, <span class="st">"ovarian"</span>, <span class="st">"sarcoma"</span>, <span class="st">"leukemia"</span>,</span>
<span>        <span class="st">"esophageal"</span>, <span class="st">"neuroblastoma"</span><span class="op">)</span>,</span>
<span>    units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<ul>
<li>Applied to CT genes not frequently activated in CCLE cell lines,
more than 95% of cell lines are not expressing each genes. The lack of
expression is very clear.</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">not_frequently_activated</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">CT_genes</span>,</span>
<span>    <span class="va">percent_of_negative_CCLE_cell_lines</span> <span class="op">&gt;=</span> <span class="fl">95</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/CCLE_expression.html">CCLE_expression</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">not_frequently_activated</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"lung"</span>, <span class="st">"skin"</span>, <span class="st">"bile_duct"</span>, <span class="st">"bladder"</span>, <span class="st">"colorectal"</span>,</span>
<span>        <span class="st">"lymphoma"</span>, <span class="st">"uterine"</span>, <span class="st">"myeloma"</span>, <span class="st">"kidney"</span>,</span>
<span>        <span class="st">"pancreatic"</span>, <span class="st">"brain"</span>, <span class="st">"gastric"</span>, <span class="st">"breast"</span>, <span class="st">"bone"</span>,</span>
<span>        <span class="st">"head_and_neck"</span>, <span class="st">"ovarian"</span>, <span class="st">"sarcoma"</span>, <span class="st">"leukemia"</span>,</span>
<span>        <span class="st">"esophageal"</span>, <span class="st">"neuroblastoma"</span><span class="op">)</span>,</span>
<span>    units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="ct_correlated_genes">
<code>CT_correlated_genes()</code><a class="anchor" aria-label="anchor" href="#ct_correlated_genes"></a>
</h4>
<p>A function that use expression data from all CCLE cell lines and
returns genes correlated (or anti-correlated) with specified CT
gene.</p>
<p>Here with MAGEA3, we can see all genes (CT or not) whose expression
is correlated with it, like MAGEA6.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CT_correlated_genes.html">CT_correlated_genes</a></span><span class="op">(</span><span class="st">"MAGEA3"</span>, <span class="fl">0.3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 4 rows containing missing values (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 162 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="tcga_expression">
<code>TCGA_expression()</code><a class="anchor" aria-label="anchor" href="#tcga_expression"></a>
</h4>
<p>Allows to visualise gene expression in cancer samples from TCGA. We
can thus for example compare with activation in CCLE cell lines.</p>
<ul>
<li>Applied to CT genes frequently activated in CCLE cell lines</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TCGA_expression.html">TCGA_expression</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">frequently_activated</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    tumor <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>    units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">## 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">## setting TRUE/FALSE to it.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Set `ht_opt$message = FALSE` to turn off this message.</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<ul>
<li>Applied to CT genes not frequently activated in CCLE cell lines</li>
</ul>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TCGA_expression.html">TCGA_expression</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">not_frequently_activated</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    tumor <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>    units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">## 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">## setting TRUE/FALSE to it.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Set `ht_opt$message = FALSE` to turn off this message.</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>When visualising only one tumor, there is a separation between
tumoral and peritumoral tissue.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TCGA_expression.html">TCGA_expression</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">frequently_activated</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    tumor <span class="op">=</span> <span class="st">"LUAD"</span>,</span>
<span>    units <span class="op">=</span> <span class="st">"log_TPM"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="methylation-analysis">Methylation analysis<a class="anchor" aria-label="anchor" href="#methylation-analysis"></a>
</h3>
<div class="section level4">
<h4 id="dac_induction">
<code>DAC_induction()</code><a class="anchor" aria-label="anchor" href="#dac_induction"></a>
</h4>
<p>Allows to visualise gene induction upon DAC treatment in a series of
cell lines. We can see the difference between CT genes regulated by
methylation or not</p>
<ul>
<li>Applied to genes controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">controlled_by_methylation</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CT_genes</span>, <span class="va">regulated_by_methylation</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/DAC_induction.html">DAC_induction</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">controlled_by_methylation</span><span class="op">$</span><span class="va">external_gene_name</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<ul>
<li>Applied to genes not controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">not_controlled_by_methylation</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">CT_genes</span>,</span>
<span>    <span class="op">!</span><span class="va">regulated_by_methylation</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/DAC_induction.html">DAC_induction</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">not_controlled_by_methylation</span><span class="op">$</span><span class="va">external_gene_name</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="normal_tissues_methylation">
<code>normal_tissues_methylation()</code><a class="anchor" aria-label="anchor" href="#normal_tissues_methylation"></a>
</h4>
<p>Gives the methylation level of all CpGs located in each promoter
region (by default defined as 1000 nt upstream TSS and 200 nt downstream
TSS) in a set of normal tissues. Can be used to analyse more precisely
the methylation of a particular promoter in normal tissues as
methylation values are given CpG by CpG.</p>
<ul>
<li>Applied to a gene controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/normal_tissues_methylation.html">normal_tissues_methylation</a></span><span class="op">(</span><span class="st">"MAGEB2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<ul>
<li>Applied to a gene not controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/normal_tissues_methylation.html">normal_tissues_methylation</a></span><span class="op">(</span><span class="st">"LIN28A"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="normal_tissues_mean_methylation">
<code>normal_tissues_mean_methylation()</code><a class="anchor" aria-label="anchor" href="#normal_tissues_mean_methylation"></a>
</h4>
<p>Gives the mean methylation level of CpGs located in each promoter
region (by default defined as 1000 nt upstream TSS and 200 nt downstream
TSS) in a set of normal tissues. When comparing genes controlled or not
by methylation, we can see a specific methylation pattern in the first
group.</p>
<ul>
<li>Applied to genes controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/normal_tissues_mean_methylation.html">normal_tissues_mean_methylation</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">controlled_by_methylation</span><span class="op">$</span><span class="va">external_gene_name</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<ul>
<li>Applied to genes not controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/normal_tissues_mean_methylation.html">normal_tissues_mean_methylation</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span></span>
<span>        <span class="va">not_controlled_by_methylation</span><span class="op">$</span><span class="va">external_gene_name</span><span class="op">)</span></span></code></pre></div>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="tcga_methylation_expression_correlation">
<code>TCGA_methylation_expression_correlation()</code><a class="anchor" aria-label="anchor" href="#tcga_methylation_expression_correlation"></a>
</h4>
<p>Shows the correlation between gene expression and promoter
methylation in TCGA samples.</p>
<ul>
<li>Applied to a gene controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TCGA_methylation_expression_correlation.html">TCGA_methylation_expression_correlation</a></span><span class="op">(</span></span>
<span>    tumor <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>    gene <span class="op">=</span> <span class="st">"TDRD1"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in is.na(cor): is.na() applied to non-(list or vector) of type</span></span>
<span><span class="co">## 'closure'</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<ul>
<li>Applied to a gene not controlled by methylation</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TCGA_methylation_expression_correlation.html">TCGA_methylation_expression_correlation</a></span><span class="op">(</span></span>
<span>    tumor <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>    gene <span class="op">=</span> <span class="st">"LIN28A"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in is.na(cor): is.na() applied to non-(list or vector) of type</span></span>
<span><span class="co">## 'closure'</span></span></code></pre>
<p><img src="CTexploreR_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>Aguet, F., Anand, S., Ardlie, K. G., Gabriel, S., Getz, G. A.,
Graubert, A., Hadley, K., Handsaker, R. E., Huang, K. H., Kashin, S.,
Li, X., MacArthur, D. G., Meier, S. R., Nedzel, J. L., Nguyen, D. T.,
Segrè, A. V., Todres, E., Balliu, B., Barbeira, A. N., … Volpi, S.
(2020). The GTEx Consortium atlas of genetic regulatory effects across
human tissues. Science. <a href="https://science.sciencemag.org/content/369/6509/1318.abstract" class="external-link uri">https://science.sciencemag.org/content/369/6509/1318.abstract</a></p>
<p>Almeida, L. G., Sakabe, N. J., deOliveira, A. R., Silva, M. C. C.,
Mundstein, A. S., Cohen, T., Chen, Y.-T., Chua, R., Gurung, S., Gnjatic,
S., Jungbluth, A. A., Caballero, O. L., Bairoch, A., Kiesler, E., White,
S. L., Simpson, A. J. G., Old, L. J., Camargo, A. A., &amp; Vasconcelos,
A. T. R. (2009). CTdatabase: a knowledge-base of high-throughput and
curated data on cancer-testis antigens. Nucleic Acids Research,
37(Database issue), D816–D819.</p>
<p>Barretina, J., Caponigro, G., Stransky, N., Venkatesan, K., Margolin,
A. A., Kim, S., Wilson, C. J., Lehár, J., Kryukov, G. V., Sonkin, D.,
Reddy, A., Liu, M., Murray, L., Berger, M. F., Monahan, J. E., Morais,
P., Meltzer, J., Korejwa, A., Jané-Valbuena, J., … Garraway, L. A.
(2012). The Cancer Cell Line Encyclopedia enables predictive modelling
of anticancer drug sensitivity. Nature, 483(7391), 603–607.</p>
<p>Cancer Genome Atlas Research Network, Weinstein, J. N., Collisson, E.
A., Mills, G. B., Shaw, K. R. M., Ozenberger, B. A., Ellrott, K.,
Shmulevich, I., Sander, C., &amp; Stuart, J. M. (2013). The Cancer
Genome Atlas Pan-Cancer analysis project. Nature Genetics, 45(10),
1113–1120.</p>
<p>Carter, J. A., Matta, B., Battaglia, J., Somerville, C., Harris, B.
D., LaPan, M., Atwal, G. S., &amp; Barnes, B. J. (2023). Identification
of pan-cancer/testis genes and validation of therapeutic targeting in
triple-negative breast cancer: Lin28a- and Siglece-based vaccination
induces anti-tumor immunity and inhibits metastasis. In bioRxiv
(p. 2023.05.09.539617).</p>
<p>Guo, J. et al. The adult human testis transcriptional cell atlas.
Cell Res. 28, 1141–1157 (2018).</p>
<p>Uhlén, M. et al. Proteomics. Tissue-based map of the human proteome.
Science 347, 1260419 (2015).</p>
<p>Thorvaldsdóttir, H., Robinson, J. T., &amp; Mesirov, J. P. (2013).
Integrative Genomics Viewer (IGV): high-performance genomics data
visualization and exploration. Briefings in Bioinformatics, 14(2),
178–192.</p>
<p>Wang, C., Gu, Y., Zhang, K., Xie, K., Zhu, M., Dai, N., Jiang, Y.,
Guo, X., Liu, M., Dai, J., Wu, L., Jin, G., Ma, H., Jiang, T., Yin, R.,
Xia, Y., Liu, L., Wang, S., Shen, B., … Hu, Z. (2016). Systematic
identification of genes with a cancer-testis expression pattern in 19
cancer types. Nature Communications, 7, 10499.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SingleCellExperiment_1.23.0 SummarizedExperiment_1.31.1</span></span>
<span><span class="co">##  [3] Biobase_2.61.0              GenomicRanges_1.53.2       </span></span>
<span><span class="co">##  [5] GenomeInfoDb_1.37.6         IRanges_2.35.2             </span></span>
<span><span class="co">##  [7] S4Vectors_0.39.2            BiocGenerics_0.47.0        </span></span>
<span><span class="co">##  [9] MatrixGenerics_1.13.1       matrixStats_1.0.0          </span></span>
<span><span class="co">## [11] CTexploreR_0.99.2           CTdata_1.1.4               </span></span>
<span><span class="co">## [13] BiocStyle_2.29.2           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] DBI_1.1.3                     bitops_1.0-7                 </span></span>
<span><span class="co">##   [3] rlang_1.1.1                   magrittr_2.0.3               </span></span>
<span><span class="co">##   [5] clue_0.3-65                   GetoptLong_1.0.5             </span></span>
<span><span class="co">##   [7] compiler_4.3.1                RSQLite_2.3.1                </span></span>
<span><span class="co">##   [9] png_0.1-8                     systemfonts_1.0.5            </span></span>
<span><span class="co">##  [11] vctrs_0.6.3                   stringr_1.5.0                </span></span>
<span><span class="co">##  [13] pkgconfig_2.0.3               shape_1.4.6                  </span></span>
<span><span class="co">##  [15] crayon_1.5.2                  fastmap_1.1.1                </span></span>
<span><span class="co">##  [17] magick_2.8.0                  dbplyr_2.3.4                 </span></span>
<span><span class="co">##  [19] XVector_0.41.1                ellipsis_0.3.2               </span></span>
<span><span class="co">##  [21] labeling_0.4.3                utf8_1.2.3                   </span></span>
<span><span class="co">##  [23] promises_1.2.1                rmarkdown_2.25               </span></span>
<span><span class="co">##  [25] ragg_1.2.6                    purrr_1.0.2                  </span></span>
<span><span class="co">##  [27] bit_4.0.5                     xfun_0.40                    </span></span>
<span><span class="co">##  [29] zlibbioc_1.47.0               cachem_1.0.8                 </span></span>
<span><span class="co">##  [31] jsonlite_1.8.7                blob_1.2.4                   </span></span>
<span><span class="co">##  [33] later_1.3.1                   DelayedArray_0.27.10         </span></span>
<span><span class="co">##  [35] interactiveDisplayBase_1.39.0 parallel_4.3.1               </span></span>
<span><span class="co">##  [37] cluster_2.1.4                 R6_2.5.1                     </span></span>
<span><span class="co">##  [39] bslib_0.5.1                   stringi_1.7.12               </span></span>
<span><span class="co">##  [41] RColorBrewer_1.1-3            jquerylib_0.1.4              </span></span>
<span><span class="co">##  [43] Rcpp_1.0.11                   bookdown_0.35                </span></span>
<span><span class="co">##  [45] iterators_1.0.14              knitr_1.44                   </span></span>
<span><span class="co">##  [47] Matrix_1.6-1.1                httpuv_1.6.11                </span></span>
<span><span class="co">##  [49] tidyselect_1.2.0              abind_1.4-5                  </span></span>
<span><span class="co">##  [51] yaml_2.3.7                    doParallel_1.0.17            </span></span>
<span><span class="co">##  [53] codetools_0.2-19              curl_5.1.0                   </span></span>
<span><span class="co">##  [55] lattice_0.21-9                tibble_3.2.1                 </span></span>
<span><span class="co">##  [57] shiny_1.7.5                   withr_2.5.1                  </span></span>
<span><span class="co">##  [59] KEGGREST_1.41.4               evaluate_0.22                </span></span>
<span><span class="co">##  [61] desc_1.4.2                    BiocFileCache_2.9.1          </span></span>
<span><span class="co">##  [63] circlize_0.4.15               ExperimentHub_2.9.1          </span></span>
<span><span class="co">##  [65] Biostrings_2.69.2             pillar_1.9.0                 </span></span>
<span><span class="co">##  [67] BiocManager_1.30.22           filelock_1.0.2               </span></span>
<span><span class="co">##  [69] foreach_1.5.2                 generics_0.1.3               </span></span>
<span><span class="co">##  [71] rprojroot_2.0.3               RCurl_1.98-1.12              </span></span>
<span><span class="co">##  [73] ggplot2_3.4.3                 BiocVersion_3.18.0           </span></span>
<span><span class="co">##  [75] munsell_0.5.0                 scales_1.2.1                 </span></span>
<span><span class="co">##  [77] xtable_1.8-4                  glue_1.6.2                   </span></span>
<span><span class="co">##  [79] tools_4.3.1                   AnnotationHub_3.9.2          </span></span>
<span><span class="co">##  [81] fs_1.6.3                      grid_4.3.1                   </span></span>
<span><span class="co">##  [83] tidyr_1.3.0                   AnnotationDbi_1.63.2         </span></span>
<span><span class="co">##  [85] colorspace_2.1-0              GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">##  [87] cli_3.6.1                     rappdirs_0.3.3               </span></span>
<span><span class="co">##  [89] textshaping_0.3.7             fansi_1.0.5                  </span></span>
<span><span class="co">##  [91] S4Arrays_1.1.6                ComplexHeatmap_2.17.0        </span></span>
<span><span class="co">##  [93] dplyr_1.1.3                   gtable_0.3.4                 </span></span>
<span><span class="co">##  [95] sass_0.4.7                    digest_0.6.33                </span></span>
<span><span class="co">##  [97] ggrepel_0.9.3                 SparseArray_1.1.12           </span></span>
<span><span class="co">##  [99] farver_2.1.1                  rjson_0.2.21                 </span></span>
<span><span class="co">## [101] memoise_2.0.1                 htmltools_0.5.6.1            </span></span>
<span><span class="co">## [103] pkgdown_2.0.7.9000            lifecycle_1.0.3              </span></span>
<span><span class="co">## [105] httr_1.4.7                    GlobalOptions_0.1.2          </span></span>
<span><span class="co">## [107] mime_0.12                     bit64_4.0.5</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Axelle Loriot, Julie Devis, Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
